driver-core/src/main/java/com/datastax/driver/core/Metadata.java

```diff
@@ -241,11 +241,15 @@ public static String quoteIfNecessary(String id) {
   /**
    * We don't need to escape an identifier if it matches non-quoted CQL3 ids ([a-z][a-z0-9_]*), and
    * if it's not a CQL reserved keyword.
+   *
+   * <p>When 'Migrating from compact storage' after DROP COMPACT STORAGE on the table, it can have a
+   * column with an empty name. For that case, we need to escape empty column name.
    */
   private static boolean needsQuote(String s) {
     // this method should only be called for C*-provided identifiers,
     // so we expect it to be non-null and non-empty.
```

Comment:
Nit: we can remove the comment here about it being expected to be "non-empty", as you have it covered in the javadoc comment above (and it's no longer true)

Reply:
adutra beat me to it!